setup.py

```diff
@@ -37,30 +62,156 @@ def build_cmake(self, ext):
+            cmake_args += [
+                f"-DLLVM_DIR={llvm_dir}",
+                f"-DMLIR_DIR={mlir_dir}",
+            ]
+            print(f"Using built LLVM from: {wave_llvm_dir}")
+            print(f"  LLVM_DIR: {llvm_dir}")
+            print(f"  MLIR_DIR: {mlir_dir}")
+
         # Clang is required on Windows, since Wave runtime uses variable-length
-        # arrays (VLAs) which not supported by MSVC
+        # arrays (VLAs) which are not supported by MSVC
         if os.name == "nt":
             cmake_args += [
-                "-G",
-                "Ninja",
                 "-DCMAKE_C_COMPILER=clang",
                 "-DCMAKE_CXX_COMPILER=clang++",
             ]
 
-        subprocess.check_call(["cmake", ext.sourcedir, *cmake_args], cwd=build_dir)
+        cmake_args += ext.cmake_args
+
+        invoke_cmake(ext.sourcedir, *cmake_args, cwd=build_dir)
 
         # Build CMake project
-        subprocess.check_call(["cmake", "--build", "."], cwd=build_dir)
+        invoke_cmake("--build", ".", "--target", "install", cwd=build_dir)
+
+    def _build_llvm(self):
+        """Build LLVM from source using the commit hash from water/llvm-sha.txt."""
+
+        # Read LLVM commit hash from water/llvm-sha.txt
+        llvm_sha_file = Path(REPO_ROOT) / "water" / "llvm-sha.txt"
+        if not llvm_sha_file.exists():
+            raise RuntimeError(
+                f"LLVM SHA file not found at {llvm_sha_file}. "
+                "Cannot build LLVM from source."
+            )
+        llvm_sha = llvm_sha_file.read_text().strip()
+
+        # Setup directories
+        water_dir = Path(REPO_ROOT) / "water"
+        llvm_dir = water_dir / "llvm-project"
+        llvm_install_dir = water_dir / "llvm-install"
+        llvm_build_dir = Path(self.build_temp) / "llvm"
+
+        # Check if LLVM is already built with the correct SHA
+        vcs_revision_file = (
+            llvm_install_dir / "include" / "llvm" / "Support" / "VCSRevision.h"
+        )
+        if llvm_install_dir.exists() and vcs_revision_file.exists():
+            # Read the installed LLVM SHA
+            vcs_content = vcs_revision_file.read_text()
+            installed_sha = None
+            for line in vcs_content.split("\n"):
+                if line.strip().startswith("#define LLVM_REVISION"):
+                    # Extract SHA from: #define LLVM_REVISION "478e45fb..."
+                    installed_sha = line.split('"')[1]
+                    break
+
+            if installed_sha == llvm_sha:
+                print(
+                    f"LLVM already built with correct SHA {llvm_sha}, skipping rebuild"
+                )
+                return llvm_install_dir
+            else:
+                print(
+                    f"LLVM SHA mismatch: installed={installed_sha}, expected={llvm_sha}"
+                )
+
+        print(f"Removing old LLVM installation...")
+        shutil.rmtree(llvm_install_dir, ignore_errors=True)
```

Comment:
Do we need to ignore errors? It looks like we may be unable to remove something, which will then may or may not get overwritten during installation, leading to a frankenmonster installation that is a nightmare to debug

Reply:
without `ignore_errors` it will fail if directory doesn't exests